contracts/Edition.sol

```diff
@@ -298,45 +299,33 @@ contract Edition is
     /// @dev Private function to mint without any access checks
     /// @return _tokenId the id of the newly minted token
     function _mintEdition(address recipient) internal returns (uint56 _tokenId) {
-        _tokenId = _mintPreFlightChecks();
+        _tokenId = _mintPreFlightChecks(1);
         _mint(recipient, _tokenId);
     }
 
 
     /// @dev Private function to safe mint without any access checks
     /// @return _tokenId the id of the newly minted token
     function _safeMintEdition(address recipient) internal returns (uint56 _tokenId) {
-        _tokenId = _mintPreFlightChecks();
+        _tokenId = _mintPreFlightChecks(1);
         _safeMint(recipient, _tokenId);
     }
 
     /// @dev Private function to batch mint without any access checks
     function _mintEditions(address[] calldata recipients)
         internal
-        returns (uint256)
+        returns (uint256 endingTokenId)
     {
         uint56 n = uint56(recipients.length);
         require(n > 0, "No recipients");
 
-        // load state into memory with a single SLOAD
-        EditionState storage _state = state;
+        endingTokenId = _mintPreFlightChecks(n);
 
         unchecked {
-            uint56 startingTokenId = _state.numberMinted + 1;
-            uint56 endingTokenId = _state.numberMinted + n;
-
-            enforceSupplyLimit(_state.editionSize, endingTokenId);
-            enforceTimeLimit(_state.endOfMintPeriod);
-
             for (uint256 i = 0; i < n; ) {
-                _safeMint(recipients[i], startingTokenId + i);
+                _safeMint(recipients[i], endingTokenId - n + i + 1);
```

Comment:
this one here got me, but i did some quick maffs for some cases and it checks out

Reply:
take a look at 5efa234, it goes down smooth imo

Reply:
ty king, you dropped this :crown: